sql/v2/runtime/test/user_defined_functions_test.go

```diff
@@ -0,0 +1,209 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package runtime_test
+
+import (
+	"io"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"testing"
+
+	cesql "github.com/cloudevents/sdk-go/sql/v2"
+	"github.com/cloudevents/sdk-go/sql/v2/function"
+	"github.com/cloudevents/sdk-go/sql/v2/parser"
+	ceruntime "github.com/cloudevents/sdk-go/sql/v2/runtime"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/binding/spec"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/test"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+var TCKFileNames = []string{
+	"user_defined_functions",
+}
+
+var TCKUserDefinedFunctions = []cesql.Function{
+	function.NewFunction(
+		"HASPREFIX",
+		[]cesql.Type{cesql.StringType, cesql.StringType},
+		nil,
+		func(event cloudevents.Event, i []interface{}) (interface{}, error) {
+			str := i[0].(string)
+			prefix := i[1].(string)
+
+			return strings.HasPrefix(str, prefix), nil
+		},
+	),
+	function.NewFunction(
+		"KONKAT",
+		[]cesql.Type{},
+		cesql.TypePtr(cesql.StringType),
+		func(event cloudevents.Event, i []interface{}) (interface{}, error) {
+			var sb strings.Builder
+			for _, v := range i {
+				sb.WriteString(v.(string))
+			}
+			return sb.String(), nil
+		},
+	),
+}
+
+type ErrorType string
+
+const (
+	ParseError              ErrorType = "parse"
+	MathError               ErrorType = "math"
+	CastError               ErrorType = "cast"
+	MissingAttributeError   ErrorType = "missingAttribute"
+	MissingFunctionError    ErrorType = "missingFunction"
+	FunctionEvaluationError ErrorType = "functionEvaluation"
+)
+
+type TckFile struct {
+	Name  string        `json:"name"`
+	Tests []TckTestCase `json:"tests"`
+}
+
+type TckTestCase struct {
+	Name       string `json:"name"`
+	Expression string `json:"expression"`
+
+	Result interface{} `json:"result"`
+	Error  ErrorType   `json:"error"`
+
+	Event          *cloudevents.Event     `json:"event"`
+	EventOverrides map[string]interface{} `json:"eventOverrides"`
+}
+
+func (tc TckTestCase) InputEvent(t *testing.T) cloudevents.Event {
+	var inputEvent cloudevents.Event
+	if tc.Event != nil {
+		inputEvent = *tc.Event
+	} else {
+		inputEvent = test.FullEvent()
+	}
+
+	// Make sure the event is v1
+	inputEvent.SetSpecVersion(event.CloudEventsVersionV1)
+
+	for k, v := range tc.EventOverrides {
+		require.NoError(t, spec.V1.SetAttribute(inputEvent.Context, k, v))
+	}
+
+	return inputEvent
+}
+
+func (tc TckTestCase) ExpectedResult() interface{} {
+	switch tc.Result.(type) {
+	case int:
+		return int32(tc.Result.(int))
+	case float64:
+		return int32(tc.Result.(float64))
+	case bool:
+		return tc.Result.(bool)
+	}
+	return tc.Result
+}
+
+func Test_functionTable_AddFunction(t *testing.T) {
```

Comment:
We tend to follow the common go test naming convention: TestPascalCase. `TestFunctionTableAddFunction`)